apps/directory/src/functions/bookmarkMapper.js

```diff
@@ -0,0 +1,133 @@
+import router from "../router";
+import { useFiltersStore } from '../stores/filtersStore'
+import { useCollectionStore } from '../stores/collectionStore';
+import { useCheckoutStore } from '../stores/checkoutStore';
+
+function setBookmark (bookmark) {
+    router.push(
+        {
+            name: router.currentRoute.name,
+            query: bookmark
+        },
+    )
+}
+
+export async function applyBookmark (watchedQuery) {
+    const checkoutStore = useCheckoutStore();
+    const collectionStore = useCollectionStore();
+    const filtersStore = useFiltersStore();
+
+    let query = watchedQuery;
+
+    if (!query) {
+        const route = router.currentRoute.value
+
+        if (!route.query) return
+
+        query = route.query
+    }
+
+    if (!query || !Object.keys(query).length > 0) return
+
+    /**  negotiator token */
+    // if (query.nToken) {
+    //   state.nToken = query.nToken
+    // }
+
+    if (query.cart) {
+        const decoded = decodeURIComponent(query.cart)
+        const cartIdString = window.atob(decoded)
+        const cartIds = cartIdString.split(',')
+
+        const missingCollections = await collectionStore.getMissingCollectionInformation(cartIds)
+        if (missingCollections && Object.keys(missingCollections).length) {
+            for (const collection of missingCollections) {
+                checkoutStore.addCollectionsToSelection({ biobank: collection.biobank, collections: [{ label: collection.name, value: collection.id }] })
+            }
+        }
+
+        /** add the beginning of history if from a link-back url */
+        // if (state.searchHistory.length === 0) {
+        //     state.searchHistory.push('Starting with a preselected list of collections')
+        // }
+    }
+
+    /** we load the filters, grab the names, so we can loop over it to map the selections */
+    const filters = Object.keys(filtersStore.facetDetails)
+
+    if (query.matchAll) {
+        const matchAllFilters = decodeURIComponent(query.matchAll).split(',')
+        for (const filterName of matchAllFilters) {
+            filtersStore.updateFilterType(filterName, "all", true)
+        }
+    }
+
+    for (const filterName of filters) {
+        if (query[filterName]) {
+
+            const filterOptions = filtersStore.filterOptionsCache[filterName];
+            let filtersToAdd = decodeURIComponent(query[filterName])
+
+            if (filterOptions) {
+                const queryValues = filtersToAdd.split(',')
+                filtersToAdd = filterOptions.filter(fo => queryValues.includes(fo.value))
+            }
+
+            filtersStore.updateFilter(filterName, filtersToAdd, true)
+        }
+    }
+}
+export function createBookmark (filters, collectionCart) {
+    const filtersStore = useFiltersStore();
+    const bookmark = {}
+    const matchAll = []
```

Comment:
You declare a const but still modify it. I know the reference doesn't change, but I'd prefer a 'let' so it's clear that the content is being modified.

Reply:
im not re-assigning. that is what a const protects for in js

Reply:
see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const

it is correct behaviour.

Reply:
I know it's correct ;) I just don't like it :p